### PR TITLE
Fix user sample thickness resetting when processing data

### PR DESCRIPTION
**Pull request by DavidFair**
**Cannot be merged before #27239**

**Description of work.**
Custom sample thicknesses were not being preserved. This fixes the issue so that if a custom one is set in the table, we respect that rather than loading the file thickness over the top.

Additionally, I've improved the warning to help users find what they did wrong, previously it just said that we failed to convert to a float in the SANS state object.

**Report to:** ISIS/Steve

**To test:**
- Download the ISIS training data
- Open the ISIS SANS GUI
- Go to the LOQ demo folder and set the user file to `maskfile.txt`
- Set the batch file to `batch.csv`
- Modify the pre-set sample thicknesses to something that isn't 1.0
- Hit load then check they are what you set
- Select the first row and hit process selected, check it's still what you set
- Hit process all and check that it's still set
- Change the sample thickness to something that isn't a number
- Hit either load / process selected / process all and check a warning is emitted
- Check that 1.0 is used instead

Fixes #27239 . 

*This does not require release notes* because this broke between releases

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/test/SANS/gui_logic/table_model_test.py [hunk 1]
```diff
@@ -131,6 +132,101 @@ def test_that_can_retrieve_sample_thickness_from_table_index_model(self):
...
+
+        logger_mock = mock.Mock()
+        table_model.logger = logger_mock
+
+        table_model.update_thickness_from_file_information(id=0, file_information=file_info_mock)
+
+        logger_mock.warning.assert_any_call(ANY)
+        self.assertEqual(3, logger_mock.warning.call_count)  # One warning for each sample dim
+
+    def test_convert_to_float(self):
```
Comment (Harrietbrown) [inline scripts/test/SANS/gui_logic/table_model_test.py]: This should be 3 tests.

**Review by Harrietbrown — CHANGES_REQUESTED** [review]
The code works, just some notes.

File: scripts/test/SANS/gui_logic/table_model_test.py [hunk 2]
```diff
@@ -80,7 +81,7 @@ def test_that_incorrect_sample_shape_reverts_to_previous_sampleshape(self):
                                                 "", "", "", "", "", "", "", "",
                                                 sample_shape="Disc")
             table_index_model.sample_shape = "not a sample shape"
-        except Exception as e:
+        except Exception:
```
[the remark this answers is not in the document]
Comment (DavidFair) [reply scripts/test/SANS/gui_logic/table_model_test.py]: This is unfortunately a really common pattern I'm slowly clearing out of SANS tests

in this case it's been highlighted after auto-formatting (as we weren't using e), I'll tidy it up anyway :+1:

**Comment by DavidFair** [comment]
Sorry @Harrietbrown I've just realised I force pushed when you (potentially) had a copy checked out

**Comment by Harrietbrown** [comment]
my local branch is no issue, But it does mean I can't restrict the changes to only the commits since last review,

**Review by Harrietbrown — APPROVED** [review]
Very good, the values reset to 1 for values that cannot be converted to a float. the failing test is unrelated to the PR

**Pull request merged**